code/core/system/proxy-camera/src/ProxyCamera.cpp

```diff
@@ -32,25 +43,99 @@ namespace proxy {
 
 using namespace std;
 using namespace odcore::base;
+using namespace odcore::data;
+using namespace odtools::recorder;
 
 ProxyCamera::ProxyCamera(const int &argc, char **argv)
-    : TimeTriggeredConferenceClientModule(argc, argv, "proxy-camera") {}
+    : TimeTriggeredConferenceClientModule(argc, argv, "proxy-camera"),
+    m_recorder(),
+    m_camera() {}
 
 ProxyCamera::~ProxyCamera() {}
 
-void ProxyCamera::setUp() {}
+void ProxyCamera::setUp() {
+            // This method will be call automatically _before_ running body().
+            if (getFrequency() < 20) {
+                cerr << endl << endl << "proxy-camera: WARNING! Running proxy-camera with a LOW frequency (consequence: data updates are too seldom and will influence your algorithms in a negative manner!) --> suggestions: --freq=20 or higher! Current frequency: " << getFrequency() << " Hz." << endl << endl << endl;
```

Comment:
@HangYin2015 This if statement was needed for CaroloCup only and needs to be removed.